src/ui/pages/SidePage/components/IncomingRequest/components/SignRequest.tsx

```diff
@@ -1,48 +1,55 @@
 import { IonText } from "@ionic/react";
-import { useMemo, useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import { i18n } from "../../../../../../i18n";
 import {
   CardDetailsAttributes,
   CardDetailsBlock,
 } from "../../../../../components/CardDetails";
 import { PageFooter } from "../../../../../components/PageFooter";
-import { PageHeader } from "../../../../../components/PageHeader";
 import { ScrollablePageLayout } from "../../../../../components/layout/ScrollablePageLayout";
 import CardanoLogo from "../../../../../assets/images/CardanoLogo.jpg";
 import { RequestProps } from "../IncomingRequest.types";
-import "./SignTransactionRequest.scss";
+import "./SignRequest.scss";
+import { PeerConnection } from "../../../../../../core/cardano/walletConnect/peerConnection";
+import { Agent } from "../../../../../../core/agent/agent";
+import { PeerConnectionMetadataRecord } from "../../../../../../core/agent/records";
 
-const SignTransactionRequest = ({
+const SignRequest = ({
   pageId,
   activeStatus,
   requestData,
   handleAccept,
   handleCancel,
 }: RequestProps) => {
-  const signTransaction = requestData.signTransaction;
+  const signRequest = requestData.signTransaction;
   const [isSigningObject, setIsSigningObject] = useState(false);
-  const ballotLogo = requestData.logo ? requestData.logo : CardanoLogo;
+  const [peerConnection, setPeerConnection] =
+    useState<PeerConnectionMetadataRecord>();
+  const logo = requestData.logo ? requestData.logo : CardanoLogo;
+
+  useEffect(() => {
+    (async () => {
+      const connectedDAppAddress =
+        PeerConnection.peerConnection.getConnectedDAppAddress();
+      const peerConnectionData =
+        await Agent.agent.peerConnectionMetadataStorage.getPeerConnectionMetadata(
+          connectedDAppAddress
+        );
+      setPeerConnection(peerConnectionData);
```

Comment:
Hmm, could we get this in the AppWrapper and pass it down via the event instead? That way it's pre-loaded. And doesn't load in case we can't find the connection